sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/CloudHsmClusterData.cs

```diff
@@ -0,0 +1,69 @@
+        public CloudHsmClusterData(AzureLocation location) : base(location)
+        {
+            Hsms = new ChangeTrackingList<CloudHsmProperties>();
+            PrivateEndpointConnections = new ChangeTrackingList<HardwareSecurityModulesPrivateEndpointConnectionData>();
+        }
+
+        /// <summary> Initializes a new instance of CloudHsmClusterData. </summary>
+        /// <param name="id"> The id. </param>
+        /// <param name="name"> The name. </param>
+        /// <param name="resourceType"> The resourceType. </param>
+        /// <param name="systemData"> The systemData. </param>
+        /// <param name="tags"> The tags. </param>
+        /// <param name="location"> The location. </param>
+        /// <param name="provisioningState"> The Cloud HSM Cluster's provisioningState. </param>
+        /// <param name="autoGeneratedDomainNameLabelScope"> The Cloud HSM Cluster's auto-generated Domain Name Label Scope. </param>
+        /// <param name="securityDomain"> Security domain properties information for Cloud HSM cluster. </param>
+        /// <param name="hsms"> An array of Cloud HSM Cluster's HSMs. </param>
+        /// <param name="publicNetworkAccess"> The Cloud HSM Cluster public network access. </param>
+        /// <param name="privateEndpointConnections"> List of private endpoint connection resources. </param>
+        /// <param name="sku"> SKU details. </param>
+        internal CloudHsmClusterData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, IDictionary<string, string> tags, AzureLocation location, ProvisioningState? provisioningState, string autoGeneratedDomainNameLabelScope, CloudHsmClusterSecurityDomainProperties securityDomain, IList<CloudHsmProperties> hsms, string publicNetworkAccess, IList<HardwareSecurityModulesPrivateEndpointConnectionData> privateEndpointConnections, CloudHsmClusterSku sku) : base(id, name, resourceType, systemData, tags, location)
+        {
+            ProvisioningState = provisioningState;
+            AutoGeneratedDomainNameLabelScope = autoGeneratedDomainNameLabelScope;
+            SecurityDomain = securityDomain;
+            Hsms = hsms;
+            PublicNetworkAccess = publicNetworkAccess;
+            PrivateEndpointConnections = privateEndpointConnections;
+            Sku = sku;
+        }
+
+        /// <summary> The Cloud HSM Cluster's provisioningState. </summary>
+        public ProvisioningState? ProvisioningState { get; set; }
+        /// <summary> The Cloud HSM Cluster's auto-generated Domain Name Label Scope. </summary>
+        public string AutoGeneratedDomainNameLabelScope { get; set; }
+        /// <summary> Security domain properties information for Cloud HSM cluster. </summary>
+        public CloudHsmClusterSecurityDomainProperties SecurityDomain { get; set; }
+        /// <summary> An array of Cloud HSM Cluster's HSMs. </summary>
+        public IList<CloudHsmProperties> Hsms { get; }
```

Comment:
This property name is a little bit weird. Can we rename it to `HsmProperties` or something similar?